src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -67,50 +67,43 @@ def _compute_mask_indices(
     Adapted from `fairseq's data_utils.py
     <https://github.com/pytorch/fairseq/blob/e0788f7007a8473a76db573985031f3c94201e79/fairseq/data/data_utils.py#L376>`__.
     """
+    torch_device = "cuda" if torch.cuda.is_available() else "cpu"
     bsz, all_sz = shape
-    mask = np.full((bsz, all_sz), False)
-
-    all_num_mask = int(
-        # add a random number for probabilistic rounding
-        mask_prob * all_sz / float(mask_length)
-        + np.random.rand()
-    )
-
-    all_num_mask = max(min_masks, all_num_mask)
-
-    mask_idcs = []
+    mask = torch.full((bsz, all_sz), False, device=torch_device)
+    min_len_unique, mask_idcs = float("Inf"), []
     padding_mask = attention_mask.ne(1) if attention_mask is not None else None
+    if padding_mask is not None:
+        sz = all_sz - padding_mask.long().sum(dim=1)
+        num_mask = (((sz * mask_prob) / mask_length) + torch.rand(bsz, device=torch_device)).int()
+    else:
+        sz = torch.full([bsz], all_sz, device=torch_device)
+        all_num_mask = ((all_sz * mask_prob) / mask_length) + torch.rand(1, device=torch_device)[0]
+        num_mask = torch.full([bsz], all_num_mask, device=torch_device).int()
+
+    num_mask = torch.clamp(num_mask, min=min_masks)
     for i in range(bsz):
-        if padding_mask is not None:
-            sz = all_sz - padding_mask[i].long().sum().item()
-            num_mask = int(
-                # add a random number for probabilistic rounding
-                mask_prob * sz / float(mask_length)
-                + np.random.rand()
-            )
-            num_mask = max(min_masks, num_mask)
-        else:
-            sz = all_sz
-            num_mask = all_num_mask
-
-        lengths = np.full(num_mask, mask_length)
-
-        if sum(lengths) == 0:
-            lengths[0] = min(mask_length, sz - 1)
-
-        min_len = min(lengths)
-        if sz - min_len <= num_mask:
-            min_len = sz - num_mask - 1
-
-        mask_idc = np.random.choice(sz - min_len, num_mask, replace=False)
-        mask_idc = np.asarray([mask_idc[j] + offset for j in range(len(mask_idc)) for offset in range(lengths[j])])
-        mask_idcs.append(np.unique(mask_idc[mask_idc < sz]))
+        curr_num_mask = int(num_mask[i])
+        curr_sz = sz[i]
+        lengths = torch.full([curr_num_mask], mask_length, device=torch_device)
+        if lengths.nelement() == 0:
+            lengths[0] = min(mask_length, curr_sz - 1)
+
+        min_len = lengths[0]
```

Comment:
why not use `torch.min(...)` here?

Reply:
`torch.min` works only on Tensors,  here we have `mask_length` and `curr_sz-1` as scalar values and thus, `torch.min` will not work.